src/core2/graph.test.js

```diff
@@ -597,14 +613,31 @@ describe("graph", () => {
     });
 
     it("throws for null or undefined dst", () => {
-      const e = (dst: any) => ({...edge(), dst});
+      const e = (dst: any) => {
+        const edge = new SimpleEdge(srcRef(), dstRef());
+        (edge: any).dst = () => dst;
+        return edge;
+      };
       expect(() => {
         newGraph().addEdge(e(null));
       }).toThrow("null");
       expect(() => {
         newGraph().addEdge(e(undefined));
       }).toThrow("undefined");
     });
+
+    it("adds edges that refer to the right graph", () => {
+      const graph = newGraph();
+      const otherGraph = newGraph();
+      const srcRef = otherGraph.ref(new FooPayload().address());
+      const edge = new SimpleEdge(srcRef, srcRef);
+      graph.addEdge(edge);
+      const edgeFromGraph = graph.edge(edge.address());
+      if (edgeFromGraph == null) {
+        throw new Error("Shocking turn of events");
+      }
+      expect(edgeFromGraph.src().graph()).toEqual(graph);
```

Comment:
This should be `toBe` instead of `toEqual`; we don’t want a deep
equality check.